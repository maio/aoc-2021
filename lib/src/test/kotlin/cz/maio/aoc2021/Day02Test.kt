package cz.maio.aoc2021

import cz.maio.ResourcesUtils.resourceLines
import org.approvaltests.Approvals
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

enum class Direction { forward, down, up }

data class Command(
    val direction: Direction,
    val value: Int
)

data class Position(
    val x: Int = 0,
    val depth: Int = 0
) {
    fun applyCommand(command: Command): Position {
        return when (command.direction) {
            Direction.forward -> copy(x = x + command.value)
            Direction.down -> copy(depth = depth + command.value)
            Direction.up -> copy(depth = depth - command.value)
        }
    }
}

data class Position2(
    val aim: Int = 0,
    val x: Int = 0,
    val depth: Int = 0
) {
    fun applyEntry(command: Command): Position2 {
        return when (command.direction) {
            Direction.forward -> copy(x = x + command.value, depth = depth + command.value * aim)
            Direction.down -> copy(aim = aim + command.value)
            Direction.up -> copy(aim = aim - command.value)
        }
    }
}

class Day02Test {
    private val entries = resourceLines("/input-02.txt").map {
        val (cmd, value) = it.split(" ")
        Command(Direction.valueOf(cmd), value.toInt())
    }

    @Test
    fun parsing() {
        Approvals.verify(entries.joinToString("\n\n"))
    }

    @Test
    fun `part 1`() {
        val finalPosition = entries.fold(Position(), Position::applyCommand)

        assertThat(finalPosition.x * finalPosition.depth).isEqualTo(2322630)
    }

    @Test
    fun `part 2`() {
        val finalPosition = entries.fold(Position2(), Position2::applyEntry)

        assertThat(finalPosition.x * finalPosition.depth).isEqualTo(2105273490)
    }
}
