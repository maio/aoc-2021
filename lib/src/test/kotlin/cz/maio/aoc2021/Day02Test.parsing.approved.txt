Command(direction=forward, value=6)

Command(direction=down, value=2)

Command(direction=forward, value=2)

Command(direction=down, value=8)

Command(direction=forward, value=3)

Command(direction=down, value=6)

Command(direction=down, value=8)

Command(direction=down, value=9)

Command(direction=forward, value=7)

Command(direction=forward, value=8)

Command(direction=down, value=9)

Command(direction=down, value=8)

Command(direction=down, value=9)

Command(direction=up, value=8)

Command(direction=forward, value=1)

Command(direction=down, value=7)

Command(direction=down, value=3)

Command(direction=forward, value=3)

Command(direction=forward, value=1)

Command(direction=down, value=3)

Command(direction=forward, value=3)

Command(direction=forward, value=1)

Command(direction=up, value=8)

Command(direction=down, value=5)

Command(direction=down, value=1)

Command(direction=forward, value=6)

Command(direction=forward, value=2)

Command(direction=up, value=9)

Command(direction=down, value=3)

Command(direction=down, value=8)

Command(direction=down, value=3)

Command(direction=down, value=3)

Command(direction=up, value=2)

Command(direction=down, value=7)

Command(direction=down, value=3)

Command(direction=up, value=5)

Command(direction=forward, value=4)

Command(direction=down, value=9)

Command(direction=forward, value=6)

Command(direction=forward, value=3)

Command(direction=forward, value=1)

Command(direction=forward, value=3)

Command(direction=down, value=2)

Command(direction=up, value=9)

Command(direction=down, value=4)

Command(direction=forward, value=6)

Command(direction=down, value=3)

Command(direction=forward, value=2)

Command(direction=down, value=2)

Command(direction=up, value=5)

Command(direction=up, value=1)

Command(direction=forward, value=3)

Command(direction=forward, value=6)

Command(direction=down, value=6)

Command(direction=forward, value=7)

Command(direction=forward, value=1)

Command(direction=down, value=3)

Command(direction=down, value=8)

Command(direction=forward, value=2)

Command(direction=down, value=7)

Command(direction=up, value=1)

Command(direction=up, value=2)

Command(direction=forward, value=5)

Command(direction=down, value=8)

Command(direction=down, value=8)

Command(direction=forward, value=9)

Command(direction=forward, value=7)

Command(direction=forward, value=2)

Command(direction=forward, value=7)

Command(direction=up, value=6)

Command(direction=up, value=9)

Command(direction=down, value=4)

Command(direction=forward, value=4)

Command(direction=forward, value=4)

Command(direction=up, value=1)

Command(direction=down, value=7)

Command(direction=forward, value=9)

Command(direction=forward, value=3)

Command(direction=down, value=6)

Command(direction=down, value=9)

Command(direction=forward, value=7)

Command(direction=forward, value=4)

Command(direction=up, value=7)

Command(direction=up, value=6)

Command(direction=up, value=8)

Command(direction=down, value=9)

Command(direction=forward, value=1)

Command(direction=down, value=1)

Command(direction=forward, value=8)

Command(direction=down, value=7)

Command(direction=forward, value=5)

Command(direction=down, value=3)

Command(direction=down, value=3)

Command(direction=down, value=8)

Command(direction=down, value=8)

Command(direction=down, value=4)

Command(direction=up, value=4)

Command(direction=forward, value=3)

Command(direction=down, value=8)

Command(direction=down, value=9)

Command(direction=up, value=3)

Command(direction=up, value=8)

Command(direction=down, value=9)

Command(direction=up, value=5)

Command(direction=forward, value=2)

Command(direction=forward, value=5)

Command(direction=forward, value=5)

Command(direction=down, value=8)

Command(direction=forward, value=9)

Command(direction=forward, value=8)

Command(direction=down, value=5)

Command(direction=down, value=9)

Command(direction=forward, value=6)

Command(direction=forward, value=2)

Command(direction=forward, value=3)

Command(direction=up, value=1)

Command(direction=forward, value=1)

Command(direction=up, value=2)

Command(direction=up, value=2)

Command(direction=forward, value=4)

Command(direction=forward, value=8)

Command(direction=forward, value=5)

Command(direction=down, value=1)

Command(direction=up, value=4)

Command(direction=forward, value=5)

Command(direction=up, value=7)

Command(direction=down, value=5)

Command(direction=down, value=5)

Command(direction=forward, value=8)

Command(direction=up, value=2)

Command(direction=down, value=7)

Command(direction=down, value=6)

Command(direction=down, value=5)

Command(direction=down, value=5)

Command(direction=down, value=1)

Command(direction=down, value=8)

Command(direction=forward, value=9)

Command(direction=forward, value=2)

Command(direction=up, value=6)

Command(direction=up, value=4)

Command(direction=down, value=8)

Command(direction=forward, value=1)

Command(direction=forward, value=2)

Command(direction=down, value=2)

Command(direction=forward, value=7)

Command(direction=forward, value=7)

Command(direction=forward, value=3)

Command(direction=forward, value=6)

Command(direction=forward, value=8)

Command(direction=down, value=3)

Command(direction=forward, value=6)

Command(direction=up, value=5)

Command(direction=down, value=3)

Command(direction=down, value=8)

Command(direction=up, value=1)

Command(direction=forward, value=1)

Command(direction=down, value=7)

Command(direction=down, value=3)

Command(direction=up, value=5)

Command(direction=forward, value=6)

Command(direction=forward, value=8)

Command(direction=forward, value=9)

Command(direction=up, value=5)

Command(direction=up, value=5)

Command(direction=up, value=5)

Command(direction=forward, value=8)

Command(direction=up, value=5)

Command(direction=down, value=6)

Command(direction=down, value=7)

Command(direction=down, value=5)

Command(direction=up, value=7)

Command(direction=up, value=1)

Command(direction=up, value=3)

Command(direction=forward, value=8)

Command(direction=up, value=9)

Command(direction=down, value=7)

Command(direction=down, value=4)

Command(direction=up, value=6)

Command(direction=up, value=8)

Command(direction=up, value=9)

Command(direction=up, value=9)

Command(direction=forward, value=5)

Command(direction=up, value=5)

Command(direction=forward, value=2)

Command(direction=forward, value=2)

Command(direction=forward, value=6)

Command(direction=up, value=2)

Command(direction=down, value=8)

Command(direction=up, value=2)

Command(direction=forward, value=5)

Command(direction=down, value=9)

Command(direction=up, value=7)

Command(direction=down, value=9)

Command(direction=forward, value=1)

Command(direction=forward, value=8)

Command(direction=up, value=1)

Command(direction=forward, value=7)

Command(direction=forward, value=2)

Command(direction=down, value=3)

Command(direction=forward, value=3)

Command(direction=forward, value=2)

Command(direction=up, value=9)

Command(direction=forward, value=4)

Command(direction=forward, value=9)

Command(direction=down, value=9)

Command(direction=forward, value=5)

Command(direction=forward, value=1)

Command(direction=forward, value=5)

Command(direction=forward, value=8)

Command(direction=up, value=5)

Command(direction=forward, value=1)

Command(direction=down, value=4)

Command(direction=up, value=8)

Command(direction=up, value=4)

Command(direction=up, value=7)

Command(direction=forward, value=4)

Command(direction=down, value=1)

Command(direction=up, value=6)

Command(direction=forward, value=6)

Command(direction=down, value=2)

Command(direction=down, value=7)

Command(direction=forward, value=4)

Command(direction=up, value=7)

Command(direction=forward, value=7)

Command(direction=forward, value=9)

Command(direction=down, value=5)

Command(direction=up, value=5)

Command(direction=forward, value=4)

Command(direction=down, value=6)

Command(direction=forward, value=1)

Command(direction=up, value=8)

Command(direction=up, value=8)

Command(direction=down, value=8)

Command(direction=down, value=7)

Command(direction=forward, value=7)

Command(direction=down, value=3)

Command(direction=forward, value=7)

Command(direction=down, value=3)

Command(direction=down, value=5)

Command(direction=down, value=4)

Command(direction=up, value=8)

Command(direction=down, value=2)

Command(direction=down, value=2)

Command(direction=up, value=5)

Command(direction=forward, value=9)

Command(direction=up, value=9)

Command(direction=forward, value=2)

Command(direction=up, value=4)

Command(direction=forward, value=4)

Command(direction=down, value=2)

Command(direction=down, value=7)

Command(direction=forward, value=7)

Command(direction=down, value=1)

Command(direction=down, value=6)

Command(direction=down, value=4)

Command(direction=forward, value=6)

Command(direction=up, value=4)

Command(direction=forward, value=4)

Command(direction=down, value=6)

Command(direction=down, value=8)

Command(direction=down, value=3)

Command(direction=forward, value=7)

Command(direction=down, value=3)

Command(direction=forward, value=7)

Command(direction=down, value=7)

Command(direction=forward, value=4)

Command(direction=up, value=9)

Command(direction=down, value=5)

Command(direction=forward, value=7)

Command(direction=forward, value=7)

Command(direction=up, value=6)

Command(direction=down, value=3)

Command(direction=forward, value=9)

Command(direction=down, value=1)

Command(direction=forward, value=4)

Command(direction=up, value=9)

Command(direction=down, value=3)

Command(direction=up, value=9)

Command(direction=down, value=5)

Command(direction=up, value=6)

Command(direction=forward, value=1)

Command(direction=forward, value=9)

Command(direction=up, value=4)

Command(direction=down, value=3)

Command(direction=forward, value=1)

Command(direction=down, value=7)

Command(direction=down, value=2)

Command(direction=forward, value=2)

Command(direction=down, value=6)

Command(direction=up, value=4)

Command(direction=down, value=4)

Command(direction=up, value=9)

Command(direction=down, value=3)

Command(direction=down, value=9)

Command(direction=down, value=4)

Command(direction=down, value=1)

Command(direction=up, value=8)

Command(direction=down, value=2)

Command(direction=up, value=1)

Command(direction=forward, value=5)

Command(direction=forward, value=9)

Command(direction=forward, value=1)

Command(direction=up, value=4)

Command(direction=forward, value=5)

Command(direction=down, value=7)

Command(direction=up, value=6)

Command(direction=down, value=3)

Command(direction=forward, value=8)

Command(direction=down, value=1)

Command(direction=down, value=5)

Command(direction=forward, value=5)

Command(direction=down, value=5)

Command(direction=down, value=7)

Command(direction=down, value=8)

Command(direction=down, value=7)

Command(direction=up, value=6)

Command(direction=forward, value=8)

Command(direction=down, value=8)

Command(direction=forward, value=6)

Command(direction=down, value=6)

Command(direction=down, value=7)

Command(direction=down, value=3)

Command(direction=forward, value=2)

Command(direction=down, value=6)

Command(direction=down, value=8)

Command(direction=down, value=7)

Command(direction=down, value=3)

Command(direction=up, value=1)

Command(direction=down, value=7)

Command(direction=forward, value=8)

Command(direction=forward, value=2)

Command(direction=forward, value=5)

Command(direction=down, value=4)

Command(direction=up, value=4)

Command(direction=forward, value=9)

Command(direction=down, value=9)

Command(direction=forward, value=6)

Command(direction=down, value=7)

Command(direction=down, value=4)

Command(direction=down, value=8)

Command(direction=up, value=9)

Command(direction=forward, value=7)

Command(direction=down, value=4)

Command(direction=forward, value=7)

Command(direction=forward, value=1)

Command(direction=forward, value=7)

Command(direction=down, value=9)

Command(direction=down, value=7)

Command(direction=forward, value=3)

Command(direction=forward, value=3)

Command(direction=forward, value=2)

Command(direction=down, value=5)

Command(direction=up, value=5)

Command(direction=forward, value=5)

Command(direction=down, value=2)

Command(direction=forward, value=7)

Command(direction=forward, value=9)

Command(direction=forward, value=7)

Command(direction=down, value=7)

Command(direction=down, value=9)

Command(direction=down, value=5)

Command(direction=forward, value=2)

Command(direction=up, value=5)

Command(direction=down, value=3)

Command(direction=forward, value=7)

Command(direction=down, value=4)

Command(direction=down, value=3)

Command(direction=up, value=5)

Command(direction=down, value=6)

Command(direction=down, value=3)

Command(direction=up, value=4)

Command(direction=forward, value=3)

Command(direction=down, value=1)

Command(direction=forward, value=6)

Command(direction=forward, value=6)

Command(direction=down, value=8)

Command(direction=forward, value=9)

Command(direction=down, value=2)

Command(direction=up, value=3)

Command(direction=down, value=4)

Command(direction=down, value=5)

Command(direction=forward, value=3)

Command(direction=down, value=9)

Command(direction=forward, value=2)

Command(direction=up, value=3)

Command(direction=up, value=4)

Command(direction=forward, value=9)

Command(direction=down, value=2)

Command(direction=forward, value=9)

Command(direction=forward, value=3)

Command(direction=down, value=4)

Command(direction=down, value=2)

Command(direction=down, value=5)

Command(direction=down, value=4)

Command(direction=forward, value=4)

Command(direction=down, value=1)

Command(direction=down, value=9)

Command(direction=down, value=2)

Command(direction=forward, value=8)

Command(direction=down, value=5)

Command(direction=forward, value=5)

Command(direction=up, value=7)

Command(direction=down, value=5)

Command(direction=down, value=2)

Command(direction=forward, value=5)

Command(direction=up, value=4)

Command(direction=down, value=5)

Command(direction=up, value=3)

Command(direction=forward, value=7)

Command(direction=down, value=9)

Command(direction=forward, value=5)

Command(direction=forward, value=2)

Command(direction=forward, value=1)

Command(direction=down, value=7)

Command(direction=down, value=9)

Command(direction=down, value=2)

Command(direction=up, value=2)

Command(direction=up, value=2)

Command(direction=up, value=4)

Command(direction=down, value=4)

Command(direction=down, value=7)

Command(direction=down, value=3)

Command(direction=forward, value=5)

Command(direction=forward, value=3)

Command(direction=up, value=6)

Command(direction=down, value=6)

Command(direction=up, value=6)

Command(direction=up, value=9)

Command(direction=forward, value=8)

Command(direction=forward, value=4)

Command(direction=up, value=3)

Command(direction=forward, value=1)

Command(direction=forward, value=2)

Command(direction=up, value=5)

Command(direction=forward, value=5)

Command(direction=forward, value=8)

Command(direction=forward, value=7)

Command(direction=forward, value=4)

Command(direction=down, value=1)

Command(direction=down, value=8)

Command(direction=down, value=1)

Command(direction=forward, value=3)

Command(direction=up, value=1)

Command(direction=forward, value=7)

Command(direction=forward, value=4)

Command(direction=down, value=8)

Command(direction=forward, value=7)

Command(direction=forward, value=9)

Command(direction=forward, value=3)

Command(direction=down, value=9)

Command(direction=down, value=9)

Command(direction=down, value=3)

Command(direction=up, value=6)

Command(direction=up, value=1)

Command(direction=down, value=4)

Command(direction=forward, value=5)

Command(direction=forward, value=4)

Command(direction=forward, value=6)

Command(direction=forward, value=8)

Command(direction=down, value=6)

Command(direction=down, value=3)

Command(direction=forward, value=5)

Command(direction=forward, value=6)

Command(direction=down, value=4)

Command(direction=down, value=2)

Command(direction=up, value=3)

Command(direction=down, value=3)

Command(direction=down, value=7)

Command(direction=down, value=5)

Command(direction=down, value=5)

Command(direction=forward, value=6)

Command(direction=down, value=4)

Command(direction=forward, value=1)

Command(direction=up, value=2)

Command(direction=forward, value=3)

Command(direction=down, value=1)

Command(direction=down, value=4)

Command(direction=down, value=9)

Command(direction=down, value=7)

Command(direction=down, value=9)

Command(direction=forward, value=9)

Command(direction=down, value=6)

Command(direction=down, value=3)

Command(direction=down, value=2)

Command(direction=down, value=5)

Command(direction=up, value=8)

Command(direction=forward, value=5)

Command(direction=forward, value=5)

Command(direction=forward, value=4)

Command(direction=up, value=5)

Command(direction=forward, value=1)

Command(direction=down, value=9)

Command(direction=down, value=1)

Command(direction=up, value=5)

Command(direction=forward, value=8)

Command(direction=forward, value=6)

Command(direction=forward, value=5)

Command(direction=down, value=1)

Command(direction=up, value=5)

Command(direction=down, value=8)

Command(direction=up, value=7)

Command(direction=down, value=8)

Command(direction=down, value=2)

Command(direction=down, value=3)

Command(direction=forward, value=2)

Command(direction=up, value=4)

Command(direction=down, value=6)

Command(direction=up, value=6)

Command(direction=down, value=3)

Command(direction=down, value=7)

Command(direction=up, value=3)

Command(direction=forward, value=4)

Command(direction=down, value=3)

Command(direction=forward, value=4)

Command(direction=up, value=9)

Command(direction=forward, value=5)

Command(direction=down, value=2)

Command(direction=forward, value=7)

Command(direction=forward, value=5)

Command(direction=up, value=3)

Command(direction=up, value=2)

Command(direction=forward, value=2)

Command(direction=down, value=8)

Command(direction=down, value=1)

Command(direction=down, value=3)

Command(direction=up, value=5)

Command(direction=down, value=4)

Command(direction=forward, value=4)

Command(direction=down, value=1)

Command(direction=forward, value=9)

Command(direction=down, value=3)

Command(direction=down, value=7)

Command(direction=down, value=4)

Command(direction=down, value=4)

Command(direction=forward, value=7)

Command(direction=up, value=5)

Command(direction=forward, value=4)

Command(direction=down, value=8)

Command(direction=up, value=4)

Command(direction=forward, value=6)

Command(direction=down, value=1)

Command(direction=up, value=4)

Command(direction=forward, value=4)

Command(direction=down, value=6)

Command(direction=up, value=5)

Command(direction=up, value=1)

Command(direction=forward, value=2)

Command(direction=down, value=5)

Command(direction=forward, value=8)

Command(direction=forward, value=6)

Command(direction=down, value=8)

Command(direction=down, value=7)

Command(direction=down, value=7)

Command(direction=down, value=1)

Command(direction=forward, value=5)

Command(direction=forward, value=7)

Command(direction=forward, value=7)

Command(direction=forward, value=7)

Command(direction=up, value=3)

Command(direction=forward, value=9)

Command(direction=forward, value=1)

Command(direction=down, value=9)

Command(direction=forward, value=4)

Command(direction=up, value=8)

Command(direction=forward, value=1)

Command(direction=forward, value=5)

Command(direction=forward, value=4)

Command(direction=down, value=2)

Command(direction=forward, value=4)

Command(direction=forward, value=9)

Command(direction=forward, value=3)

Command(direction=down, value=1)

Command(direction=forward, value=4)

Command(direction=forward, value=9)

Command(direction=forward, value=5)

Command(direction=down, value=5)

Command(direction=down, value=5)

Command(direction=forward, value=7)

Command(direction=down, value=3)

Command(direction=forward, value=4)

Command(direction=down, value=6)

Command(direction=forward, value=7)

Command(direction=down, value=2)

Command(direction=down, value=1)

Command(direction=down, value=5)

Command(direction=forward, value=4)

Command(direction=forward, value=9)

Command(direction=down, value=4)

Command(direction=forward, value=2)

Command(direction=down, value=8)

Command(direction=up, value=5)

Command(direction=down, value=9)

Command(direction=forward, value=8)

Command(direction=down, value=3)

Command(direction=up, value=6)

Command(direction=down, value=2)

Command(direction=down, value=4)

Command(direction=forward, value=4)

Command(direction=up, value=2)

Command(direction=down, value=4)

Command(direction=down, value=4)

Command(direction=up, value=7)

Command(direction=down, value=6)

Command(direction=forward, value=4)

Command(direction=down, value=7)

Command(direction=forward, value=3)

Command(direction=down, value=1)

Command(direction=up, value=1)

Command(direction=down, value=2)

Command(direction=down, value=6)

Command(direction=down, value=4)

Command(direction=up, value=3)

Command(direction=down, value=6)

Command(direction=up, value=2)

Command(direction=down, value=6)

Command(direction=forward, value=3)

Command(direction=down, value=9)

Command(direction=forward, value=5)

Command(direction=down, value=5)

Command(direction=down, value=9)

Command(direction=down, value=9)

Command(direction=down, value=7)

Command(direction=forward, value=9)

Command(direction=forward, value=8)

Command(direction=forward, value=9)

Command(direction=up, value=9)

Command(direction=forward, value=7)

Command(direction=forward, value=4)

Command(direction=forward, value=4)

Command(direction=up, value=5)

Command(direction=forward, value=2)

Command(direction=down, value=1)

Command(direction=up, value=9)

Command(direction=forward, value=2)

Command(direction=forward, value=7)

Command(direction=forward, value=1)

Command(direction=down, value=9)

Command(direction=forward, value=9)

Command(direction=up, value=8)

Command(direction=up, value=1)

Command(direction=up, value=7)

Command(direction=up, value=7)

Command(direction=down, value=5)

Command(direction=forward, value=2)

Command(direction=forward, value=8)

Command(direction=forward, value=6)

Command(direction=down, value=7)

Command(direction=forward, value=1)

Command(direction=down, value=9)

Command(direction=down, value=4)

Command(direction=down, value=4)

Command(direction=down, value=1)

Command(direction=up, value=7)

Command(direction=forward, value=4)

Command(direction=forward, value=6)

Command(direction=up, value=5)

Command(direction=forward, value=2)

Command(direction=down, value=9)

Command(direction=down, value=7)

Command(direction=forward, value=1)

Command(direction=forward, value=2)

Command(direction=down, value=5)

Command(direction=forward, value=3)

Command(direction=forward, value=8)

Command(direction=forward, value=6)

Command(direction=forward, value=3)

Command(direction=forward, value=2)

Command(direction=down, value=1)

Command(direction=forward, value=1)

Command(direction=forward, value=1)

Command(direction=forward, value=3)

Command(direction=down, value=9)

Command(direction=up, value=9)

Command(direction=down, value=9)

Command(direction=down, value=6)

Command(direction=forward, value=7)

Command(direction=down, value=6)

Command(direction=forward, value=9)

Command(direction=down, value=9)

Command(direction=down, value=7)

Command(direction=down, value=1)

Command(direction=down, value=9)

Command(direction=up, value=9)

Command(direction=down, value=6)

Command(direction=forward, value=9)

Command(direction=down, value=6)

Command(direction=forward, value=3)

Command(direction=down, value=8)

Command(direction=up, value=5)

Command(direction=forward, value=5)

Command(direction=forward, value=8)

Command(direction=up, value=3)

Command(direction=down, value=8)

Command(direction=up, value=6)

Command(direction=forward, value=4)

Command(direction=down, value=2)

Command(direction=forward, value=6)

Command(direction=down, value=9)

Command(direction=forward, value=6)

Command(direction=forward, value=4)

Command(direction=forward, value=9)

Command(direction=forward, value=3)

Command(direction=down, value=2)

Command(direction=down, value=4)

Command(direction=forward, value=5)

Command(direction=down, value=9)

Command(direction=up, value=7)

Command(direction=forward, value=4)

Command(direction=up, value=1)

Command(direction=forward, value=1)

Command(direction=down, value=6)

Command(direction=forward, value=3)

Command(direction=forward, value=7)

Command(direction=forward, value=2)

Command(direction=forward, value=2)

Command(direction=down, value=5)

Command(direction=down, value=9)

Command(direction=down, value=3)

Command(direction=down, value=5)

Command(direction=up, value=3)

Command(direction=forward, value=1)

Command(direction=down, value=2)

Command(direction=down, value=4)

Command(direction=down, value=1)

Command(direction=up, value=9)

Command(direction=up, value=5)

Command(direction=up, value=1)

Command(direction=down, value=1)

Command(direction=up, value=9)

Command(direction=down, value=5)

Command(direction=up, value=3)

Command(direction=up, value=3)

Command(direction=down, value=7)

Command(direction=forward, value=4)

Command(direction=down, value=6)

Command(direction=forward, value=2)

Command(direction=forward, value=7)

Command(direction=forward, value=4)

Command(direction=down, value=2)

Command(direction=forward, value=6)

Command(direction=forward, value=2)

Command(direction=down, value=3)

Command(direction=up, value=3)

Command(direction=up, value=9)

Command(direction=forward, value=9)

Command(direction=forward, value=9)

Command(direction=forward, value=6)

Command(direction=down, value=8)

Command(direction=down, value=1)

Command(direction=forward, value=9)

Command(direction=up, value=1)

Command(direction=down, value=6)

Command(direction=forward, value=6)

Command(direction=up, value=5)

Command(direction=forward, value=2)

Command(direction=forward, value=6)

Command(direction=down, value=9)

Command(direction=forward, value=1)

Command(direction=forward, value=8)

Command(direction=down, value=8)

Command(direction=forward, value=4)

Command(direction=forward, value=7)

Command(direction=up, value=6)

Command(direction=up, value=1)

Command(direction=forward, value=7)

Command(direction=forward, value=3)

Command(direction=forward, value=2)

Command(direction=down, value=4)

Command(direction=down, value=7)

Command(direction=down, value=7)

Command(direction=down, value=1)

Command(direction=down, value=6)

Command(direction=forward, value=1)

Command(direction=down, value=9)

Command(direction=up, value=9)

Command(direction=up, value=9)

Command(direction=down, value=2)

Command(direction=down, value=2)

Command(direction=forward, value=5)

Command(direction=up, value=2)

Command(direction=forward, value=7)

Command(direction=up, value=5)

Command(direction=down, value=9)

Command(direction=forward, value=7)

Command(direction=forward, value=2)

Command(direction=down, value=8)

Command(direction=up, value=1)

Command(direction=down, value=5)

Command(direction=forward, value=6)

Command(direction=down, value=8)

Command(direction=down, value=7)

Command(direction=forward, value=4)

Command(direction=up, value=2)

Command(direction=down, value=8)

Command(direction=forward, value=2)

Command(direction=down, value=5)

Command(direction=down, value=4)

Command(direction=down, value=9)

Command(direction=down, value=1)

Command(direction=down, value=9)

Command(direction=down, value=6)

Command(direction=down, value=3)

Command(direction=forward, value=1)

Command(direction=forward, value=6)

Command(direction=up, value=1)

Command(direction=up, value=1)

Command(direction=up, value=9)

Command(direction=down, value=2)

Command(direction=down, value=2)

Command(direction=forward, value=5)

Command(direction=down, value=3)

Command(direction=forward, value=4)

Command(direction=down, value=3)

Command(direction=down, value=7)

Command(direction=down, value=7)

Command(direction=forward, value=4)

Command(direction=up, value=3)

Command(direction=forward, value=4)

Command(direction=down, value=3)

Command(direction=forward, value=8)

Command(direction=forward, value=1)

Command(direction=up, value=2)

Command(direction=up, value=1)

Command(direction=forward, value=1)

Command(direction=down, value=6)

Command(direction=down, value=1)

Command(direction=down, value=3)

Command(direction=forward, value=7)

Command(direction=down, value=7)

Command(direction=forward, value=4)

Command(direction=forward, value=5)

Command(direction=forward, value=3)

Command(direction=down, value=5)

Command(direction=forward, value=9)

Command(direction=forward, value=5)

Command(direction=down, value=7)

Command(direction=forward, value=6)

Command(direction=down, value=4)

Command(direction=down, value=4)

Command(direction=down, value=9)

Command(direction=down, value=3)

Command(direction=up, value=9)

Command(direction=forward, value=7)

Command(direction=down, value=7)

Command(direction=forward, value=6)

Command(direction=down, value=2)

Command(direction=down, value=9)

Command(direction=forward, value=4)

Command(direction=forward, value=1)

Command(direction=forward, value=4)

Command(direction=down, value=5)

Command(direction=forward, value=7)

Command(direction=down, value=9)

Command(direction=down, value=8)

Command(direction=forward, value=9)

Command(direction=forward, value=1)

Command(direction=down, value=9)

Command(direction=forward, value=6)

Command(direction=up, value=5)

Command(direction=forward, value=9)

Command(direction=down, value=1)

Command(direction=down, value=5)

Command(direction=forward, value=4)

Command(direction=forward, value=5)

Command(direction=forward, value=8)

Command(direction=down, value=5)

Command(direction=forward, value=9)

Command(direction=down, value=6)

Command(direction=down, value=2)

Command(direction=up, value=4)

Command(direction=up, value=8)

Command(direction=forward, value=3)

Command(direction=forward, value=4)

Command(direction=down, value=3)

Command(direction=forward, value=4)

Command(direction=up, value=6)

Command(direction=forward, value=3)

Command(direction=forward, value=8)

Command(direction=forward, value=7)

Command(direction=down, value=1)

Command(direction=down, value=9)

Command(direction=down, value=8)

Command(direction=down, value=8)

Command(direction=down, value=1)

Command(direction=forward, value=9)

Command(direction=up, value=4)

Command(direction=down, value=5)

Command(direction=forward, value=7)

Command(direction=down, value=8)

Command(direction=down, value=3)

Command(direction=forward, value=9)

Command(direction=down, value=5)

Command(direction=forward, value=7)

Command(direction=forward, value=2)

Command(direction=down, value=4)

Command(direction=forward, value=2)

Command(direction=forward, value=7)

Command(direction=down, value=6)

Command(direction=forward, value=7)

Command(direction=down, value=2)

Command(direction=forward, value=9)

Command(direction=down, value=9)

Command(direction=forward, value=8)

Command(direction=forward, value=8)

Command(direction=down, value=6)

Command(direction=forward, value=7)

Command(direction=down, value=8)

Command(direction=forward, value=7)

Command(direction=forward, value=3)

Command(direction=down, value=1)

Command(direction=up, value=8)

Command(direction=down, value=5)

Command(direction=down, value=6)

Command(direction=up, value=5)

Command(direction=forward, value=5)

Command(direction=forward, value=5)

Command(direction=up, value=5)

Command(direction=up, value=3)

Command(direction=up, value=7)

Command(direction=down, value=6)

Command(direction=forward, value=8)

Command(direction=forward, value=4)

Command(direction=down, value=2)

Command(direction=up, value=5)

Command(direction=forward, value=8)

Command(direction=down, value=6)

Command(direction=forward, value=4)

Command(direction=forward, value=2)

Command(direction=up, value=8)

Command(direction=down, value=8)

Command(direction=down, value=5)

Command(direction=down, value=4)

Command(direction=forward, value=9)

Command(direction=forward, value=9)

Command(direction=forward, value=6)

Command(direction=forward, value=6)

Command(direction=down, value=3)

Command(direction=up, value=1)

Command(direction=down, value=4)

Command(direction=down, value=8)

Command(direction=down, value=9)

Command(direction=down, value=1)

Command(direction=forward, value=3)

Command(direction=forward, value=1)

Command(direction=down, value=9)

Command(direction=down, value=3)

Command(direction=down, value=7)

Command(direction=forward, value=6)

Command(direction=forward, value=9)

Command(direction=down, value=8)

Command(direction=down, value=8)

Command(direction=forward, value=6)

Command(direction=forward, value=1)

Command(direction=down, value=3)

Command(direction=forward, value=1)

Command(direction=down, value=8)

Command(direction=down, value=3)

Command(direction=down, value=9)

Command(direction=up, value=1)

Command(direction=forward, value=6)

Command(direction=up, value=2)

Command(direction=down, value=3)

Command(direction=forward, value=4)

Command(direction=forward, value=2)

Command(direction=up, value=2)

Command(direction=down, value=5)

Command(direction=forward, value=1)

Command(direction=down, value=3)

Command(direction=forward, value=9)

Command(direction=forward, value=4)

Command(direction=forward, value=6)

Command(direction=down, value=3)

Command(direction=forward, value=7)

Command(direction=down, value=6)

Command(direction=up, value=3)

Command(direction=up, value=7)

Command(direction=up, value=5)

Command(direction=down, value=4)

Command(direction=forward, value=4)

Command(direction=up, value=1)

Command(direction=forward, value=7)

Command(direction=up, value=9)

Command(direction=forward, value=3)

Command(direction=up, value=1)

Command(direction=down, value=3)

Command(direction=down, value=4)

Command(direction=forward, value=4)

Command(direction=up, value=3)

Command(direction=down, value=6)

Command(direction=down, value=9)

Command(direction=down, value=6)

Command(direction=forward, value=4)

Command(direction=down, value=9)

Command(direction=down, value=6)

Command(direction=forward, value=4)

Command(direction=forward, value=3)

Command(direction=down, value=3)

Command(direction=up, value=7)

Command(direction=down, value=9)

Command(direction=forward, value=8)